lib/app-express.js

```diff
@@ -16,6 +37,48 @@ module.exports = function (app) {
     return next();
   });
 
+  app.get('/hostedOneClick', (req, res) => {
```

Comment:
I'm finding this pretty hard to follow. I wonder, should we create some single purpose functions for some in a different file?

Reply:
What happens if `hostedOneClick?template=init` is called more than once?

Reply:
these get fired if you're initing and it's already been done.  

``` javascript
if (template === 'init'){
  res.render('pages/customError', {
    customError : 'You\'re setting the template to init, but the app is already got its credentials loaded'
  });
}
```

Reply:
Here's an important question that just came to me.  

Why does your version of the the app (BYOhub) need oauth key/secret stored in it, if users are using their own hubs to sign in and generate orgs off of?

Reply:
Because they're using my Connected App.